opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/config/EncryptionConfig.java

```diff
@@ -0,0 +1,46 @@
+package de.adorsys.opba.protocol.facade.config;
+
+import com.google.common.io.Resources;
+import com.google.crypto.tink.Aead;
+import com.google.crypto.tink.CleartextKeysetHandle;
+import com.google.crypto.tink.JsonKeysetReader;
+import com.google.crypto.tink.KeysetHandle;
+import com.google.crypto.tink.aead.AeadConfig;
+import lombok.SneakyThrows;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.nio.file.Paths;
+import java.security.Provider;
+import java.security.SecureRandom;
+import java.security.Security;
+
+@Configuration
+public class EncryptionConfig {
+    public static final SecureRandom SECURE_RANDOM = new SecureRandom();
+    public static final int SALT_LENGTH = 8;
+    public static final int ITER_COUNT = 1024;
+    public static final String ALGO = "PBEWithSHA256And256BitAES-CBC-BC";
+    @Value("${security.provider.name}")
+    String providerName;
+
+    @Bean
+    @SneakyThrows
+    public Aead systemAeadConfig() {
+        AeadConfig.register();
+        String path = Paths.get(Resources.getResource("example-keyset.json").toURI()).toAbsolutePath().toString();
+        KeysetHandle systemKeysetHandle = CleartextKeysetHandle.read(JsonKeysetReader.withPath(path));
+        return systemKeysetHandle.getPrimitive(Aead.class);
+    }
+
+    @Bean
+    public Provider securityProvider() {
```

Comment:
Better to wrap Provider into some class (to avoid accidential clashes with beans of same type)
i.e. 
```
class FacadeSecurityProvider extends Provider {

  @Delegate
   private final Provider provider;
}
```